Library/Homebrew/utils.rb

```diff
@@ -253,24 +257,54 @@ def curl(*args)
   safe_system curl, *args
 end
 
-def puts_columns(items, star_items = [])
+def puts_columns(items, highlight = [])
   return if items.empty?
 
-  if star_items && star_items.any?
-    items = items.map { |item| star_items.include?(item) ? "#{item}*" : item }
+  unless $stdout.tty?
+    puts items
+    return
   end
 
-  if $stdout.tty?
-    # determine the best width to display for different console sizes
-    console_width = `/bin/stty size`.chomp.split(" ").last.to_i
-    console_width = 80 if console_width <= 0
-    max_len = items.reduce(0) { |max, item| l = item.length ; l > max ? l : max }
-    optimal_col_width = (console_width.to_f / (max_len + 2).to_f).floor
-    cols = optimal_col_width > 1 ? optimal_col_width : 1
-
-    IO.popen("/usr/bin/pr -#{cols} -t -w#{console_width}", "w") { |io| io.puts(items) }
+  # TTY case: If possible, output using multiple columns.
+  console_width = Tty.width
+  console_width = 80 if console_width <= 0
+  max_len = items.max_by(&:length).length
+  col_gap = 2 # number of spaces between columns
+  gap_str = " " * col_gap
+  cols = (console_width + col_gap) / (max_len + col_gap)
+  cols = 1 if cols < 1
+  if ENV["HOMEBREW_PUTS_COLUMNS_ROW_MAJOR"]
```

Comment:
Are you referring to `ENV["HOMEBREW_PUTS_COLUMNS_ROW_MAJOR"]`? If yes: It was easy enough to implement both [row-major and column-major order](https://en.wikipedia.org/wiki/Row-major_order) in the output, so I figured I make it easy for people to test both variants. To quote from [my previous comment](#issuecomment-145188663):

> - If environment variable `HOMEBREW_PUTS_COLUMNS_ROW_MAJOR` is set, put items in row-major order (new alternative). If unset, use column-major order (as before).
>   - *Note:* I implemented both variants for testing purposes. I will remove one of them once there's a consensus on the preferred item arrangement.